### PR TITLE
Poolers with simple tests

### PR DESCRIPTION
**Pull request by arampacha**

### CODE REVIEW

File: src/open_clip/hf_model.py [hunk 1]
```diff
@@ -91,3 +137,9 @@ def lock(self, unlocked_layers:int=0, freeze_layer_norm:bool=True):
         for n, p in self.transformer.named_parameters():
             if True: #mb optional LayerNorm params etc.
                 p.requires_grad = False
+
+# temporary for fast testing
```
Comment (iejMac) [inline src/open_clip/hf_model.py]: keeping this in?

File: src/open_clip/hf_model.py [hunk 2]
```diff
@@ -3,45 +3,91 @@
...
+
+    def forward(self, x:BaseModelOutput, attention_mask:TensorType):
+        masked_output = x.last_hidden_state * attention_mask.unsqueeze(-1)        
+        return masked_output.sum(dim=1) / attention_mask.sum(-1, keepdim=True)
+
+@register_pooler
+class MaxPooler(nn.Module):
+    "Max pooling"
+
+    def forward(self, x:BaseModelOutput, attenstion_mask:TensorType):
```
Comment (iejMac) [inline src/open_clip/hf_model.py]: attention_mask typo

File: src/open_clip/hf_model.py [hunk 3]
```diff
@@ -3,45 +3,91 @@
...
+
+    def __init__(self) -> None:
+        super().__init__()
+
+    def forward(self, x:BaseModelOutput, attention_mask:TensorType):
+        masked_output = x.last_hidden_state * attention_mask.unsqueeze(-1)        
+        return masked_output.sum(dim=1) / attention_mask.sum(-1, keepdim=True)
+
+@register_pooler
+class MaxPooler(nn.Module):
```
Comment (iejMac) [inline src/open_clip/hf_model.py:42]: missing __init__. If not needed here then why does MeanPooler have one?

File: src/open_clip/hf_model.py (hunk 1 above)
Comment (arampacha) [reply src/open_clip/hf_model.py]: guess we might have different development routines, so let's keep it separate, removed

File: src/open_clip/hf_model.py (hunk 3 above)
Comment (arampacha) [reply src/open_clip/hf_model.py:42]: good point, removed `__ini__` from `MeanPooler` for consistency, I've also removed `DummyPooler` - don't think we'll need it

File: src/open_clip/hf_model.py (hunk 2 above)
Comment (arampacha) [reply src/open_clip/hf_model.py]: thanks

**Pull request merged**